source/common/filesystem/posix/directory_iterator_impl.cc

```diff
@@ -55,7 +55,11 @@ FileType DirectoryIteratorImpl::fileType(const std::string& full_path) const {
 
   const Api::SysCallIntResult result = os_sys_calls_.stat(full_path.c_str(), &stat_buf);
   if (result.rc_ != 0) {
-    throw EnvoyException(fmt::format("unable to stat file: '{}'", full_path));
+    if (errno == ENOENT)
+      // Special case, a symlink is broken, target couldn't be stat'ed
+      if (::lstat(full_path.c_str(), &stat_buf) == 0 && S_ISLNK(stat_buf.st_mode))
+        return FileType::Regular;
```

Comment:
braces around all if statements please, here and below. This should be caught by clang-tidy but clang-tidy is currently broken. @lizan @derekargueta any ETA on fixing that?

Reply:
Ack, reviewed and fixed throughout. Leaving unresolved for the open question of clang-tidy.